File: Programming Skill 1/509.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        dp = [0,1]
        for i in range(2,n+1):
            dp.append(dp[i-1]+dp[i-2])
        return dp[n]

File: Solution with Js/509.js
var fib = function(n){
    function fib_seq(n){
        if(n===0) return n;
        if(n===1) return n;

        return fib_seq(n-1) + fib(n-2);
    }
    return fib_seq(n);
};

n = 10;
for (let i=0;i<=n;i++){
    console.log(fib(i))
}

